### PR TITLE
Devdocs: Remove experimental WordPress components

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

Remove devdocs for experimental `@wordpress/components`. We should not encourage usage of experimental APIs.

See p4TIVU-9xW-p2#comment-10759

#### Testing instructions

* Devdocs should build and compile.
* Experimental components should not be available:  https://calypso.live/devdocs/wordpress-components-gallery?branch=remove/devdocs-wp-experimental-apis
* Lint should pass. 
* Full lint for this rule `npx eslint-nibble --ext .js,.jsx,.ts,.tsx --no-interactive --rule wpcalypso/no-unsafe-wp-apis .` should report no errors.

### CODE REVIEW

File: client/devdocs/design/wordpress-components-gallery/autocomplete.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ const AutocompleteExample = () => {
 		<div>
 			<Autocomplete record={ value } onChange={ setValue } completers={ autocompleters }>
 				{ ( { isExpanded, listBoxId, activeId, onKeyDown } ) => (
-					<InputControl
+					<input
```
Comment (sirreal) [inline client/devdocs/design/wordpress-components-gallery/autocomplete.tsx]: I think this broke 😕

**Comment by sirreal** [comment]
@saramarcondes or @tyxla I think y'all both worked on this. I think it's close, may need a bit of polish making sure the component examples still work (Autocomplete?). Could you help to bring it home?

**Comment by tyxla** [comment]
> @saramarcondes or @tyxla I think y'all both worked on this. I think it's close, may need a bit of polish making sure the component examples still work (Autocomplete?). Could you help to bring it home?

Sure thing! Also, Sara is AFK for this and next week, so I'm happy to help with this.

FWIW the Gutenberg `<Autocomplete />` component isn't feature-complete and from my experiments to use it here and there, it doesn't work very well. Also, @ellatrix mentioned [here](https://github.com/WordPress/gutenberg/issues/10542#issuecomment-618283128) that it's considered experimental and not recommended for external use:

> Unfortunately, this component wasn't really meant to be exported for use in plugins and I highly discourage using this component and see it as experimental. I guess we'll (have to) keep it around in its current form, but at some point I don't expect it to be used by Gutenberg core anymore.

So my suggestion is to completely remove it from the WordPress components gallery for the time being. I've removed it in ce435032b52d4c475ee28aa0b478a85f58fd539a.

**Comment by tyxla** [comment]
There are possibly some stable components we could add to the gallery, but this is work for another time and another PR. 

I've given the testing instructions a shot, too, everything works as expected.

From my perspective, this looks good to go already. 

I see it's marked as Draft @sirreal, did you mean to add/change anything else before marking it as ready to review?

**Comment by sirreal** [comment]
Thanks! I wanted to sort out Autocomplete, so otherwise this is good to go I'd say.

**Review by tyxla — APPROVED** [review]
LGTM 🚢

**Pull request merged**